featurebyte/routes/historical_feature_table/controller.py

```diff
@@ -129,3 +134,64 @@ async def update_historical_feature_table(
         )
         assert isinstance(table, HistoricalFeatureTableModel)
         return table
+
+    async def preview_feature(
```

Comment:
The logic in this function seems to overlap with this method: https://github.com/featurebyte/featurebyte/blob/2c444d16eeea72e1e17a9d15182b60f5afd31610/featurebyte/routes/common/base_materialized_table.py#L210-L250

Is it possible to refactor and reuse?